packages/babel-parser/test/fixtures/typescript/arrow-function/generic-tsx-babel-7/output.json

```diff
@@ -1,6 +1,9 @@
 {
   "type": "File",
   "start":0,"end":79,"loc":{"start":{"line":1,"column":0},"end":{"line":2,"column":18}},
+  "errors": [
+    "SyntaxError: Single type parameter T should have a trailing comma. Example usage: <T,>. (2:17)"
```

Comment:
Uh, this made me realize that `this.state.startLoc` doesn't work because we didn't just finish parsing the type parameters, but the whole arrow. I guess we can go back to `{ at: invalidSingleType.end }` :slightly_frowning_face: 
```

Reply:
You right, I actually realized that something wrong with that. Taking it back so and will update the PR

Reply:
@nicolo-ribaudo but now it says 2:2 actually it should be 2:3 right?

Reply:
2:2 is way better than 2:17 :shrug: 

If we still want to report 2:3 we could store `this.state.startLoc` after the `typeParameters = this.tsParseTypeParameters();` call a few lines above.

Reply:
I did something different, lets see if it passes from tests, and if a valid usage for you also?

Reply:
@nicolo-ribaudo do you think it is okay?

Reply:
Hi @JLHwung  can you give a last review for the PR? thanks